src/cargo/core/resolver/mod.rs

```diff
@@ -999,6 +992,11 @@ fn find_candidate(
             }
         }
 
+        let (candidate, has_another) = frame
```

Comment:
Double checking: commit says this is moved out of the loop but its moved to later in the loop, right?

Reply:
Yes. The structure here is a little weird. The "hot path" is `while ... continue` which eventually exits by `while ... return`. Actually getting to the end of the loop only happens once in a resolution.

Reply:
I pushed another commit reorganizing the loop to be more normal.